File: docs/07-archive/legacy-repos/ebs_bo/src/bo/schemas/series.py
from pydantic import BaseModel


class SeriesCreate(BaseModel):
    series_name: str
    competition_id: int
    year: int
    begin_at: str
    end_at: str
    time_zone: str = "UTC"
    currency: str = "USD"
    country_code: str | None = None


class SeriesUpdate(BaseModel):
    series_name: str | None = None
    year: int | None = None
    begin_at: str | None = None
    end_at: str | None = None
    time_zone: str | None = None
    currency: str | None = None
    country_code: str | None = None


class SeriesRead(BaseModel):
    series_id: int
    competition_id: int
    series_name: str
    year: int
    begin_at: str
    end_at: str
    time_zone: str
    currency: str
    country_code: str | None
    created_at: str
    updated_at: str

File: team1-frontend/lib/models/entities/series.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'series.freezed.dart';
part 'series.g.dart';

@freezed
class Series with _$Series {
  const factory Series({
    @JsonKey(name: 'seriesId') required int seriesId,
    @JsonKey(name: 'competitionId') required int competitionId,
    @JsonKey(name: 'seriesName') required String seriesName,
    required int year,
    @JsonKey(name: 'beginAt') required String beginAt,
    @JsonKey(name: 'endAt') required String endAt,
    @JsonKey(name: 'imageUrl') String? imageUrl,
    @JsonKey(name: 'timeZone') required String timeZone,
    required String currency,
    @JsonKey(name: 'countryCode') String? countryCode,
    @JsonKey(name: 'isCompleted') @Default(false) bool isCompleted,
    @JsonKey(name: 'isDisplayed') @Default(true) bool isDisplayed,
    @JsonKey(name: 'isDemo') @Default(false) bool isDemo,
    required String source,
    @JsonKey(name: 'syncedAt') String? syncedAt,
    @JsonKey(name: 'createdAt') required String createdAt,
    @JsonKey(name: 'updatedAt') required String updatedAt,
  }) = _Series;

  factory Series.fromJson(Map<String, dynamic> json) =>
      _$SeriesFromJson(json);
}
